lib/services/ngsi/subscriptionService.js

```diff
@@ -0,0 +1,211 @@
+ * @param {Function} callback   Callback to be called when the subscription handler ends.
+ * @return {Function}          Returns a request handler for the given data.
+ */
+function createSubscriptionHandler(device, triggers, callback) {
+    return function(error, response, body) {
+        if (error || !body) {
+            logger.debug(
+                context,
+                'Transport error found subscribing device with id [%s] to entity [%s]', device.id, device.name);
+
+            callback(error);
+
+        } else if (response.statusCode !== 200 && response.statusCode !== 201) {
+            logger.debug(context, 'Unknown error subscribing device with id [%s] to entity [%s]: $s',
+                response.statusCode);
+
+            callback(new errors.EntityGenericError(device.name, device.type, {
+                details: body
+            }, response.statusCode));
+
+        } else if (body && body.orionError) {
+            logger.debug(
+                context,
+                'Orion found subscribing device with id [%s] to entity [%s]: %s',
+                device.id, device.name, body.orionError);
+
+            callback(new errors.BadRequest(body.orionError.details));
+        } else {
+            if (!device.subscriptions) {
+                device.subscriptions = [];
+            }
+
+            device.subscriptions.push({
+                id: body.subscribeResponse.subscriptionId,
+                triggers: triggers
+            });
+
+            deviceRegistry.update(device, callback);
+        }
+    };
+}
+
+/**
+ * Makes a subscription for the given device's entity, triggered by the given attributes. The contents of the
+ * notification can be selected using the "content" array (that can be left blank to notify the complete entity).
+ *
+ * @param {Object} device       Object containing all the information about a particular device.
+ * @param {Object} triggers     Array with the names of the attributes that would trigger the subscription
+ * @param {Object} content      Array with the names of the attributes to retrieve in the notification.
+ */
+function subscribe(device, triggers, content, callback) {
+    var options = {
+        method: 'POST',
+        headers: {
+            'fiware-service': device.service,
+            'fiware-servicepath': device.subservice
+        },
+        json: {
+            entities: [
+                {
+                    type: device.type,
+                    isPattern: 'false',
+                    id: device.id
+                }
+            ],
+            reference: config.providerUrl + '/notify',
+            duration: config.deviceRegistrationDuration || 'P1M',
+            notifyConditions: [
+                {
+                    type: 'ONCHANGE',
+                    condValues: triggers
+                }
+            ],
+            throttling: config.throttling || 'PT5S'
```

Comment:
I don't have a strong opinion on that matter. Let's discuss it offline and I will remove the default value if it seems problematic.